modules/components/prop-types/component-instance-prop-type.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Elementor\Modules\Components\PropTypes;
+
+use Elementor\Modules\AtomicWidgets\PropTypes\Base\Plain_Prop_Type;
+use Elementor\Modules\AtomicWidgets\PropTypes\Primitives\Number_Prop_Type;
+use Elementor\Plugin;
+use Elementor\Modules\Components\Documents\Component;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Component_Instance_Prop_Type extends Plain_Prop_Type {
+	private $component_overridable_props = null;
+
+	public static function get_key(): string {
+		return 'component-instance';
+	}
+
+	protected function validate_value( $value ): bool {
+		if ( ! is_array( $value ) ) {
+			return false;
+		}
+
+		$is_valid_structure = (
+			isset( $value['component_id'] ) &&
+			is_numeric( $value['component_id'] ) &&
+			( isset( $value['overrides'] ) ? is_array( $value['overrides'] ) : true )
+		);
+
+		if ( ! $is_valid_structure ) {
+			return false;
+		}
+
+		if ( ! isset( $value['overrides'] ) ) {
+			return true;
+		}
+
+		$sanitized_component_id = (int) $value['component_id'];
```

Comment:
```suggestion
		$sanitized_component_id = $value['component_id'];
```

You already checked it above